js/stringProcessing/getSentences.js

```diff
@@ -1,16 +1,122 @@
-var cleanText = require( "../stringProcessing/cleanText.js" );
+var stripSpaces = require( "../stringProcessing/stripSpaces.js" );
+
 var filter = require( "lodash/filter" );
 var isEmpty = require( "lodash/isEmpty" );
+var isUndefined = require( "lodash/isUndefined" );
+var difference = require( "lodash/difference" );
+var forEach = require( "lodash/forEach" );
+var isNaN = require( "lodash/isNaN" );
+
+/**
+ * Checks if the period is followed with a whitespace. If not, it is no ending of a sentence.
+ *
+ * @param {string} text The text to split in sentences
+ * @param {number} index The current index to look for.
+ * @returns {boolean} True if it doesn't match a whitespace.
+ */
+var inValidateOnWhiteSpace = function( text, index ) {
+	return text.substring( index + 1 ).match( /\s/ ) === null;
+};
+
+/**
+ * Checks if the character in the next sentence is a capital letter or a number. If not, this period is not the end
+ * of a sentence.
+ *
+ * @param {string} text The text to split in sentences
+ * @param {Array} indices The array with indices of periods in the text.
+ * @param {number} i The current iterator.
+ * @returns {boolean} False if it doesn't match a capital.
+ */
+var inValidateOnCapital = function( text, indices, i ) {
+	var firstChar = text.substring( indices[ i ] + 2, indices[ i  ] + 3 );
```

Comment:
Please comment this line. Also add the redundant space in the second `indices[ i  ]`.